### PR TITLE
Verify and mirror external dependencies and self-consistent package versions

### PR DESCRIPTION
**Pull request by natemcmaster**
Part of https://github.com/aspnet/CoreCLR/issues/271 - make the mirror/gatekeeper part of aspnet/Universe
Also part of https://github.com/aspnet/Coherence-Signed/issues/648 - simplifying build stages

Changes:
 - Move version verification from aspnet/Coherence to this repo
 - Add the complete list of external dependencies and where to get them
 - Add /t:Push to make push a separate action
 - Generate the 'Partners' lineup from this build instead of our gatekeeper build

### CODE REVIEW

File: build/artifacts.props [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+    <PackageArtifact Include="Microsoft.AspNetCore.DataProtection.AzureStorage" Category="ship" Metapackage="true" />
+    <PackageArtifact Include="Microsoft.AspNetCore.DataProtection.Extensions" Category="ship" Metapackage="true" />
+    <PackageArtifact Include="Microsoft.AspNetCore.DataProtection.Redis" Category="ship" />
+    <PackageArtifact Include="Microsoft.AspNetCore.DataProtection.SystemWeb" Category="ship" />
+    <PackageArtifact Include="Microsoft.AspNetCore.Diagnostics" Category="ship" Metapackage="true" />
+    <PackageArtifact Include="Microsoft.AspNetCore.Diagnostics.Abstractions" Category="ship" Metapackage="true" />
+    <PackageArtifact Include="Microsoft.AspNetCore.Diagnostics.Elm" Category="noship" />
+    <PackageArtifact Include="Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore" Category="ship" Metapackage="true" />
+    <PackageArtifact Include="Microsoft.AspNetCore.Diagnostics.Identity.Service" Category="ship" Metapackage="true" />
+    <PackageArtifact Include="Microsoft.AspNetCore.Dispatcher" Category="ship" Metapackage="true" />
```
Comment (JunTaoLuo) [inline build/artifacts.props]: This is marked as exclude in the current packages.csv. I'm assuming we can finally sunset that file after we move to this system? Can we update the new packages creation wiki to reflect this as well?

File: build/artifacts.props [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+    <PackageArtifact Include="Microsoft.AspNetCore.Dispatcher.Abstractions" Category="ship" Metapackage="true" />
```
Comment (JunTaoLuo) [inline build/artifacts.props]: also exclude.

File: build/artifacts.props [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+    <PackageArtifact Include="Microsoft.Extensions.HashCodeCombiner.Sources" Category="noship" />
+    <PackageArtifact Include="Microsoft.Extensions.Hosting.Abstractions" Category="ship" Metapackage="true" />
+    <PackageArtifact Include="Microsoft.Extensions.Hosting" Category="ship" Metapackage="true" />
+    <PackageArtifact Include="Microsoft.Extensions.Identity.Core" Category="ship" Metapackage="true" />
+    <PackageArtifact Include="Microsoft.Extensions.Identity.Stores" Category="ship" Metapackage="true" />
+    <PackageArtifact Include="Microsoft.Extensions.Localization" Category="ship" Metapackage="true" />
+    <PackageArtifact Include="Microsoft.Extensions.Localization.Abstractions" Category="ship" Metapackage="true" />
+    <PackageArtifact Include="Microsoft.Extensions.Logging" Category="ship" Metapackage="true" />
+    <PackageArtifact Include="Microsoft.Extensions.Logging.Abstractions" Category="ship" Metapackage="true" />
+    <PackageArtifact Include="Microsoft.Extensions.Logging.Analyzers" Category="noship" Metapackage="true" />
```
Comment (JunTaoLuo) [inline build/artifacts.props]: noship, exclude

Comment (JunTaoLuo) [reply build/artifacts.props]: Hold on, this is marked as noship, include in packages.csv? @pakrym I'm assuming this was a typo and should be noship, exclude?

Comment (natemcmaster) [reply build/artifacts.props]: Good catch. I just copied what was in packages.csv. I didn't validate the Metapackages="true" attribute against what is actually in the metapackage at the moment.

File: build/artifacts.props (hunk 2 above)
Comment (natemcmaster) [reply build/artifacts.props]: This has to ship. Microsoft.AspNetCore.Routing depends on it, and this is a shipping package. A shipping package cannot depend on a non-ship package. cc @rynowak

File: build/artifacts.props (hunk 1 above)
Comment (natemcmaster) [reply build/artifacts.props]: Yes, we'll take care of updating docs in https://github.com/aspnet/Coherence-Signed/issues/669

**Comment by natemcmaster** [comment]
We should also add validation that anything with Metapackage="true" is Category="ship"

File: build/artifacts.props (hunk 2 above)
Comment (JunTaoLuo) [reply build/artifacts.props]: Hmm I guess it's an error in the packages.csv then?

File: build/tasks/RepoTasks.tasks [hunk 4]
```diff
@@ -4,7 +4,9 @@
   </PropertyGroup>
 
   <UsingTask TaskName="RepoTasks.CalculateBuildGraph" AssemblyFile="$(_RepoTaskAssembly)" />
+  <UsingTask TaskName="RepoTasks.CopyPackagesToSplitFolders" AssemblyFile="$(_RepoTaskAssembly)" />
   <UsingTask TaskName="RepoTasks.PinVersions" AssemblyFile="$(_RepoTaskAssembly)" />
   <UsingTask TaskName="RepoTasks.ReadPackageIdentity" AssemblyFile="$(_RepoTaskAssembly)" />
```
Comment (JunTaoLuo) [inline build/tasks/RepoTasks.tasks]: Looks like you removed the cs source for this so I guess we can remove it here as well? Same with ReadPackageDependencies.

**Comment by JunTaoLuo** [comment]
Looks reasonable since it looks like it's mostly back porting what we had in the patch branches. I think this is ready as long as it works once the artifacts.props mismatches are resolved.

File: build/artifacts.props [hunk 5]
```diff
@@ -0,0 +1,239 @@
+<Project>
+
+  <ItemDefinitionGroup>
+    <PackageArtifact>
+      <Metapackage>false</Metapackage>
+    </PackageArtifact>
+  </ItemDefinitionGroup>
+
+  <ItemGroup>
```
Comment (muratg) [inline build/artifacts.props]: RIP packages.csv. 😢

File: build/artifacts.props (hunk 3 above)
Comment (pakrym) [reply build/artifacts.props]: Yeah, noship exclude.

File: build/tasks/RepoTasks.tasks (hunk 4 above)
Comment (natemcmaster) [reply build/tasks/RepoTasks.tasks]: Update.

**Comment by natemcmaster** [comment]
:up: :date:

**Pull request merged**